packages/notifi-frontend-client/lib/client/NotifiFrontendClient.ts

```diff
@@ -855,63 +663,51 @@ export class NotifiFrontendClient {
         'Sign message params and configuration must have the same blockchain',
       );
     }
-    switch (signMessageParams.walletBlockchain) {
-      case 'ETHEREUM':
-      case 'BERACHAIN':
-      case 'POLYGON':
-      case 'ARBITRUM':
-      case 'AVALANCHE':
-      case 'BINANCE':
-      case 'BASE':
-      case 'BLAST':
-      case 'CELO':
-      case 'MANTLE':
-      case 'LINEA':
-      case 'SCROLL':
-      case 'MANTA':
-      case 'MONAD':
-      case 'EVMOS':
-      case 'THE_ROOT_NETWORK':
-      case 'OPTIMISM': {
-        const { walletPublicKey, tenantId } = this
-          ._configuration as NotifiConfigWithPublicKey;
-        const signedMessage = `${SIGNING_MESSAGE}${walletPublicKey}${tenantId}${timestamp.toString()}`;
-        const messageBuffer = new TextEncoder().encode(signedMessage);
 
-        const signedBuffer = await signMessageParams.signMessage(messageBuffer);
-        const signature = normalizeHexString(
-          Buffer.from(signedBuffer).toString('hex'),
+    if (isUsingUnmaintainedBlockchain(signMessageParams)) {
+      // Happens for chains that we discontinued support for
+      // e.g. Acala, EVMOS
+      throw new Error(
+        `Unsupported blockchain: ${signMessageParams.walletBlockchain}. Please contact us if you beleive this is an error or if you need support for a new blockchain.`,
+      );
+    }
+
+    if (isUsingCosmosBlockchain(signMessageParams)) {
+      const { signatureBase64, signedMessage } =
+        await signMessageParams.signMessage(signMessageParams.message);
+      return { signature: signatureBase64, signedMessage };
+    } else if (isUsingAptosBlockchain(signMessageParams)) {
+      const { signatureHex, signedMessage } =
+        await signMessageParams.signMessage(
+          SIGNING_MESSAGE_WITHOUT_NONCE,
+          signMessageParams.nonce,
         );
+      return { signature: signatureHex, signedMessage };
+    } else if (isUsingEvmBlockchain(signMessageParams)) {
```

Comment:
In existing version, there are two cases to login `INJECTIVE` chain in `_authenticate` method, one under EVM, one under COSMOS. But this change will restrict users only using EVM. It might break the existing injective users.